user.py

```diff
@@ -1,24 +1,51 @@
 import re
+
+
 class User:
-    Groups = {' ':0,'+':1,'☆':1,'%':2,'@':3,'*':3.1,'&':4,'#':5,'~':6}
+    """ Container class for basic user information collected from rooms.
+     
+    This information consists of user.id, user.rank, and user.name. user.id is
+    a format-removed id of user.name with only a-z lowercase and 0-9 present.
+
+    user.rank contain the auth level of the user, as a single character string of
+    either ' ', +, ☆, %, @, *, &, #, or ~. Note that ☆ is only relevant for 
+    battle rooms. 
+    
+    To compare groups against each other self.Groups have the information required 
+    when used like: User.Groups[user.rank] for a numeric value.
+
+    Lastly, user.name is the unaltered name as seen in the chat rooms, and can be 
+    used for things like replying. Comparison between users should make use of 
+    user.id since users can change their frequently.
+    
+    Attributes:
+        Groups: map string to int, ranks precedence of user ranks by symbols.
+        name: string, username.
+        id: string, simplified unique username.
+        rank: string, user rank.
+        owner: Bool, is this you.
+    """
+    Groups = {' ': 0, '+': 1, '☆': 1, '%': 2, '@': 3, '*': 3.1, '&': 4, '#': 5, '~': 6}
+
     @staticmethod
     def compareRanks(rank1, rank2):
         try:
             return User.Groups[rank1] >= User.Groups[rank2]
-        except:
-            if not rank1 in User.Groups:
+        except KeyError:
+            if rank1 not in User.Groups:
                 print('{rank} is not a supported usergroup'.format(rank = rank1))
-            if not rank2 in User.Groups:
+            if rank2 not in User.Groups:
                 print('{rank} is not a supported usergroup'.format(rank = rank2))
             return False
 
-    def __init__(self, name, rank = ' ', owner = False):
+    def __init__(self, name, rank=' ', owner=False):
         self.name = name
         self.id = re.sub(r'[^a-zA-z0-9]', '', name).lower()
         self.rank = rank
         self.owner = owner
 
     def hasRank(self, rank):
         return self.owner or User.compareRanks(self.rank, rank)
+
```

Comment:
https://github.com/QuiteQuiet/PokemonShowdownBot/blob/master/room.py#L42-L54

I've generally put spaces between method unless they have a lot in common or affect the same system. It's only /mostly/ arbitrary.